lab2_synthesis.tcl

```diff
@@ -0,0 +1,78 @@
+################################################################################
+# DESIGN COMPILER:  Logic Synthesis Tool                                       #
+################################################################################
+remove_design -all
+
+# Add search paths for our technology libs.
+set search_path "$search_path . ./verilog /w/apps2/public.2/tech/synopsys/32-28nm/SAED32_EDK/lib/stdcell_rvt/db_nldm" 
+set target_library "saed32rvt_ff1p16vn40c.db saed32rvt_ss0p95v125c.db"
+set link_library "* saed32rvt_ff1p16vn40c.db saed32rvt_ss0p95v125c.db dw_foundation.sldb"
+set synthetic_library "dw_foundation.sldb"
+
+# Define work path (note: The work path must exist, so you need to create a folder WORK first)
+define_design_lib WORK -path ./WORK
+#define_design_lib work -path ./work
+
+set alib_library_analysis_path �./alib-52/�
+
+# Read the gate-level verilog files
+
+read_file {M216A_TopModule.v latency.v least_strip.v p1_reg.v height_to_id.v strip_id_to_y.v} -autoread -format verilog -top MyTopModule

```

Comment:
This command throws error because the filepath and the top module is incorrect.
idk how the synthesis still ran and produced results; my guess is that it just synthesized the library.
```suggestion
read_file {src/M216A_TopModule.v src/latency.v src/least_strip.v src/p1_reg.v src/height_to_id.v src/strip_id_to_y.v} -autoread -format verilog -top M216A_TopModule
```

Reply:
What's the difference between `read_file` command used here vs `analyze` used in lab2/lab3

Reply:
i tried using `analyze` function, but it didn't work on my side. I searched google and it spits out 'readfile' function as an alternative.